modules/ethereumNode.js

```diff
@@ -140,20 +149,20 @@ class EthereumNode extends EventEmitter {
     }
 
 
-    restart(newType, newNetwork) {
+    restart(newType, newNetwork, newMode) {
         return Q.try(() => {
             if (!this.isOwnNode) {
                 throw new Error('Cannot restart node since it was started externally');
             }
 
-            log.info('Restart node', newType, newNetwork);
+            log.info('Restart node..', newType, newNetwork, newMode);
 
             return this.stop()
                 .then(() => {
                     Windows.loading.show();
                 })
                 .then(() => {
-                    return this._start(newType || this.type, newNetwork || this.network);
+                    return this._start(newType || this.type, newNetwork || this.network, newMode || this.nodeMode);
```

Comment:
Whats this `this.nodeMode` ? i never see this variable created,shouldnt it be `this.mode` or something?